### PR TITLE
terminate > kill, display spinner while waiting for VMs to spin up/down

### PR DESCRIPTION
**Pull request by depombo**

### CODE REVIEW

File: avm/src/vm/deploy.rs [hunk 1]
```diff
@@ -114,16 +114,20 @@ pub async fn vm_health(app_id: &str, deploy_key: &str) {
...
-    Ok(_) => println!("Killing app with id {} if it exists...\n", app_id),
-    Err(err) => println!("Killing app with id {} failed. Error: {}", app_id, err)
-  }
+  let sp = SpinnerBuilder::new(format!("Terminating app {} if it exists", app_id)).start();
+  let resp = post(format!("{}/v1/terminate", URL), body).await;
+  let res = match resp {
+    Ok(_) => format!("Terminated app {} succesfully!", app_id),
+    Err(err) => format!("Failed to terminate app {}. Error {}", app_id, err),
+  };
+  sp.update(format!(""));
```
Comment (cdmistman) [inline avm/src/vm/deploy.rs:128]: is this needed?

File: avm/src/vm/deploy.rs [hunk 2]
```diff
@@ -201,7 +206,7 @@ pub async fn status() {
 
   let mut data: Vec<Vec<&dyn Display>> = vec![];
   for app in &mut apps {
-    data.push(vec![&app.id, &app.status, &app.url, &app.cloudProvider, &app.cloudAlias]);
+    data.push(vec![&app.id, &app.url, &app.cloudProvider, &app.cloudAlias]);
```
Comment (dfellis) [inline avm/src/vm/deploy.rs:209]: Why does the `status` subcommand no longer have a status field?

Comment (depombo) [reply avm/src/vm/deploy.rs:209]: "info" is probably more accurate. Now that we wait for vm instances to spin up and down, the status is always "running". When we have SSH access or store the status of the cluster we will probably want to do something more elaborate to provide an actual "status". We could require the user to provide a `health` endpoint that we check through the main HTTPS url though, but if the user doesn't have that endpoint the app would show up as `unavailable`

Comment (depombo) [reply avm/src/vm/deploy.rs:208]: Discussed offline we decided to go with calling the command `info` for now

File: avm/src/vm/deploy.rs (hunk 1 above)
Comment (depombo) [reply avm/src/vm/deploy.rs]: removed!

**Review by dfellis — APPROVED** [review]

**Review by cdmistman — APPROVED** [review]

File: avm/src/vm/deploy.rs [hunk 3]
```diff
@@ -88,7 +88,11 @@ async fn post(url: String, body: Value) -> Result<String, Box<dyn Error>> {
   let mut resp = client.request(req).await?;
   let data = hyper::body::to_bytes(resp.body_mut()).await?;
   let str = String::from_utf8(data.to_vec())?;
```
Comment (cdmistman) [inline avm/src/vm/deploy.rs:90]: I *just* noticed that you're getting a name clash since `str` is a primitive type. Maybe use `data_str`?

Comment (depombo) [reply avm/src/vm/deploy.rs:90]: Sure, will address that in the next PR which I already started working on.

**Pull request merged**